File: Leetcode/26-Remove-Duplicates-from-Sorted-Array/RemoveDuplicatesFromSortedArray.java
public class RemoveDuplicatesFromSortedArray {
    public int removeDuplicates(int[] nums) {
        int i =0;
        if (nums == null || nums.length <1) {
            return i;
        }
        
        int j = i+1;
        int cur = nums[i];
        while (j<nums.length) {
            if (nums[j] != cur) {
                i++;
                nums[i] = nums[j];
                cur = nums[i];
            } 
            j++;
        }
        return i+1;
    }
}

File: Leetcode/26-Remove-Duplicates-from-Sorted-Array/RemoveDuplicatesFromSortedArray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    var i = 0
    if (!nums || nums.length < 1) {
        return i
    }
    var j = i+1
    var cur = nums[i]
    while (j < nums.length) {
        if (nums[j] != cur) {
            i++
            nums[i] = nums[j]
            cur = nums[i];
        }
        j++
    }
    return i+1
};


